docker-compose.yaml

```diff
@@ -96,24 +42,81 @@ services:
     container_name: supply-chain-server-default
     expose:
       - 3000
-      - 28015
-      - 4004
     ports:
-      - "3000:3000"
+      - '8020:3000'
     depends_on:
       - validator
-      - rethinkdb
-    command: "bash -c \" \
-        set -x && \
-        DB_HOST=rethinkdb npm run init && \
-        VALIDATOR_URL=tcp://validator:4004 DB_HOST=rethinkdb node index.js
-    \""
+      - rethink
+    command: |
+      bash -c "
+        set -x &&
+        DB_HOST=rethink npm run init &&
+        VALIDATOR_URL=tcp://validator:4004 DB_HOST=rethink node index.js
+      "
 
-  rethinkdb:
+  rethink:
     image: rethinkdb
     expose:
       - 8080
       - 28015
     ports:
-      - "18080:8080"
-      - "28015:28015"
+      - '8021:8080'
+      - '28020:28015'
+
+  validator:
+    image: hyperledger/sawtooth-validator:1.0
+    container_name: sawtooth-validator-default
+    expose:
+      - 4004
```

Comment:
We should probably have all exposed ports have the same prefix (280xx for example) so it's easy to determine they don't conflict with other apps which we have given different prefixes.

Reply:
With Marketplace it maintains the initial 2-3 digits of whatever the default is:
- TCP (validator): 40xx
- HTPP (clients, API, etc): 80xx
- Database connection (rethink): 280xx

I was sticking to this scheme. No reason we couldn't simplify it further to 280xx (or whatever) for everything though, if people prefer.

Reply:
Expose ports aren't reachable by the host machine, so they don't need to be changed.  Only ports under the "ports:" setting would need to be exposed, which in this case would be the Web App, and maybe the rethink console.

Reply:
Although the comment is on a line with an `expose` port I took it to be about the ports actually exposed to the host machine with `ports` a few lines below.